### ISSUE TITLE
Config var from attached addon is removed on consecutive script run

### ISSUE THREAD

**Issue by bukowskiadam** [Schibsted-Tech-Polska/HeroIn#6]
Steps to reproduce:

1. Create app X, with addon that adds config var (e.g. quotaguardstatic)
```
{
    stage: 'production',
    spec: {
...
        config_vars: {
            LOG_LEVEL: 'debug',
        },
        addons: {
            quotaguardstatic: {
                plan: 'quotaguardstatic:starter',
            },
        },
    },
}
```
In config vars you should have `LOG_LEVEL` and `QUOTAGUARDSTATIC_URL`
2. Create app Y and attach addon from previous app
```
{
    stage: 'production',
    spec: {
...
        config_vars: {
            LOG_LEVEL: 'debug',
        },
        addons: {
            quotaguardstatic: {
                plan: 'quotaguardstatic:starter',
                name: <unique name generated from previous app>,
            },
        },
    },
}
```
After first run of deploy script there should be both variables, but after second one there will be no `QUOTAGUARDSTATIC_URL` in config vars.

**Comment by kjarmicki** [comment]
Whoa, it seems like it's actually a bug (or _very weird feature_) in Heroku.

I've created a parent app with an addon and connected that to a child app, then CURLed their addons info:
`curl -n https://api.heroku.com/apps/test-parent-with-addon/addons -H "Accept: application/vnd.heroku+json; version=3"`
->
```json
[
  {
    "actions":[],
    "config_vars":[
      "REDIS_URL"
    ],
    "created_at":"2017-11-02T15:29:23Z",
    "id":"40a433e3-cc96-45f7-b7ce-5b5509acf17a",
    "name":"redis-aerodynamic-70286",
    "addon_service":{
      "id":"5bbf672c-07f6-49c2-9c16-f1dcb96784db",
      "name":"heroku-redis"
    },
    "plan":{
      "id":"67756275-86b8-4edc-80c5-d543f9df7d44",
      "name":"heroku-redis:hobby-dev"
    },
    "app":{
      "id":"087a2a84-8905-4b5d-b76e-521222be6373",
      "name":"test-parent-with-addon"
    },
    "provider_id":"resource20827580@heroku.com",
    "state":"provisioned",
    "updated_at":"2017-11-02T15:35:52Z",
    "web_url":"https://addons-sso.heroku.com/apps/087a2a84-8905-4b5d-b76e-521222be6373/addons/40a433e3-cc96-45f7-b7ce-5b5509acf17a"
  }
]
```
As you can see `config_vars` are populated just fine, but now, when we ask for child addons:
`curl -n https://api.heroku.com/apps/test-child-with-addon/addons -H "Accept: application/vnd.heroku+json; version=3"`
->
```json
[
  {
    "actions":[],
    "config_vars":[],
    "created_at":"2017-11-02T15:29:23Z",
    "id":"40a433e3-cc96-45f7-b7ce-5b5509acf17a",
    "name":"redis-aerodynamic-70286",
    "addon_service":{
      "id":"5bbf672c-07f6-49c2-9c16-f1dcb96784db",
      "name":"heroku-redis"
    },
    "plan":{
      "id":"67756275-86b8-4edc-80c5-d543f9df7d44",
      "name":"heroku-redis:hobby-dev"
    },
    "app":{
      "id":"087a2a84-8905-4b5d-b76e-521222be6373",
      "name":"test-parent-with-addon"
    },
    "provider_id":"resource20827580@heroku.com",
    "state":"provisioned",
    "updated_at":"2017-11-02T15:35:52Z",
    "web_url":"https://addons-sso.heroku.com/apps/087a2a84-8905-4b5d-b76e-521222be6373/addons/40a433e3-cc96-45f7-b7ce-5b5509acf17a"
  }
]
```
Boom, no `config_vars`! Even though the addon is attached and everything is the same, it's incorrectly listing its `config_vars` as empty.

I'm taking that to Heroku support, we'll see what they say.

**Comment by kjarmicki** [comment]
Ok, so according to Heroku this is a correct behaviour because `the add-ons endpoint only returns the config vars exposed to the owning app by this add-on`. I still think that's weird, but we'll have to work around it.
Fortunately, that should be easy enough - if the current app is not the owning app then get the addon info from the owning app. I'll implement a fix for that in the coming days.

**Comment by kjarmicki** [comment]
@bukowskiadam Fix for this issue [was implemented](https://github.com/Schibsted-Tech-Polska/HeroIn/commit/9af523bc46276a5d1ac92b1845a28032c88fb114) and released with version `1.19.6`, please check it out :)
